dll/win32/rpcrt4/rpc_async.c

```diff
@@ -158,3 +158,47 @@ RPC_STATUS WINAPI RpcAsyncCancelCall(PRPC_ASYNC_STATE pAsync, BOOL fAbortCall)
     FIXME("(%p, %s): stub\n", pAsync, fAbortCall ? "TRUE" : "FALSE");
     return RPC_S_INVALID_ASYNC_HANDLE;
 }
+
+/***********************************************************************
+ *           RpcGetAuthorizationContextForClient [RPCRT4.@]
+ * 
+ * Returns the Authz context for an RPC client.
+ * 
+ * PARAMS
+ *  ClientBinding       [I] Binding handle on the server that represents
+ *                          a binding to a client.
+ *  ImpersonateOnReturn [I] Directs the function to impersonate the client on return.
+ *  Reserved1           [I] Reserved. Must be null.
+ *  pExpirationTime     [I] Pointer to the expiration date and time of the token.
+ *  Reserved2           [I] Reserved. Must be a LUID structure with each member set to zero.
+ *  Reserved3           [I] Reserved. Must be zero.
+ *  Reserved4           [I] Reserved. Must be null.
+ *  pAuthzClientContext [I] Pointer to an AUTHZ_CLIENT_CONTEXT_HANDLE structure
+ *                          that can be passed directly to Authz functions.
+ * 
+ * RETURNS
+ *  Success: RPC_S_OK.
+ *  Failure: Any error code.
+ */
+RPC_STATUS
+WINAPI
+RpcGetAuthorizationContextForClient(RPC_BINDING_HANDLE ClientBinding,
+                                    BOOL ImpersonateOnReturn,
+                                    void * Reserved1,
+                                    PLARGE_INTEGER pExpirationTime,
+                                    LUID Reserved2,
+                                    DWORD Reserved3,
+                                    PVOID Reserved4,
+                                    PVOID *authz_client_context)
+{
+    FIXME("(%p, %s, %p, %p, %p, %d, %p, %p): stub\n",
+          ClientBinding,
+          ImpersonateOnReturn,
+          Reserved1,
+          pExpirationTime,
+          Reserved2,
+          Reserved3,
+          Reserved4,
+          authz_client_context);
+    return RPC_S_NO_CONTEXT_AVAILABLE;
```

Comment:
I'm guessing it's necessary to return failure here?  If returning `0` works to solve your problem, you could get away with just a `-stub` mark in the .spec file.

Reply:
Thanks. Adding a `-stub` in spec file indeed the same fixed the problem for me. At least "Failed to snap..." error does no longer appear for me. But I think if this function is documented, why not add a proper stub instead of just export?...

Reply:
Because adding it in code means maintaining a diff to upstream. The spec file, on the other hand, is ours.

Reply:
But if I'll send a patch in Wine, and if it will be accepted, there will no diff, right? So it will be not needed to maintain it. If this will not happen, I think I'll be able to update the diff from time to time, after each winesync.